reana_job_controller/kubernetes_job_manager.py

```diff
@@ -405,20 +422,46 @@ def add_image_pull_secrets(self):
 
         self.job["spec"]["template"]["spec"]["imagePullSecrets"] = image_pull_secrets
 
-    def add_memory_limit(self, job_spec):
-        """Add limits.memory to job accordingly."""
+    def validate_resources(self):
+        """Validate that resource requests are less than or equal to limits."""
+        if self.kubernetes_cpu_request and self.kubernetes_cpu_limit:
+            cpu_request = kubernetes_cpu_to_millicores(self.kubernetes_cpu_request)
+            cpu_limit = kubernetes_cpu_to_millicores(self.kubernetes_cpu_limit)
+            if cpu_request > cpu_limit:
+                raise REANAKubernetesRequestExceedsLimit(
```

Comment:
Cosmetics: In the worklfow logs, this error message appears as:

```
2025-04-04 08:08:13,492 | reana-workflow-engine-snakemake | MainThread | ERROR | Error submitting job gendata: Job submission error: Job submission failed.
CPU request (0.2) cannot be greater than limit (0.1). If you are overriding the values, please check the default and maximum values for requests and limits with 'reana-client info' command.
2025-04-04 08:08:13,492 | reana-workflow-engine-snakemake | MainThread | INFO | gendata job is failed. job_id: None
```

We may want to print leading "ERROR: CPU request (0.2) cannot be ..." so that if people grep for errors, they would find this line easily. 

(This can be added more generally when printing all sorts of exceptions, or just here in the message...)